lib/elixir/lib/file/stream.ex

```diff
@@ -73,7 +73,8 @@ defmodule File.Stream do
       start_fun =
         fn ->
           case :file.open(path, read_modes(modes)) do
-            {:ok, device}    -> device
+            {:ok, device}    ->
+              if :strip_bom in modes, do: strip_bom(device), else: device
```

Comment:
Is there are any best practice on "when we should split function" in Elixir source? Level of nesting, ABC complexity, etc..?